packages/esm-patient-chart-app/src/visit/visits-widget/current-visit-summary.extension.tsx

```diff
@@ -1,34 +1,26 @@
 import React from 'react';
 import { useTranslation } from 'react-i18next';
-import { InlineLoading } from '@carbon/react';
-import { ErrorState, launchWorkspace, useVisit } from '@openmrs/esm-framework';
-import { CardHeader, EmptyState } from '@openmrs/esm-patient-common-lib';
+import { launchWorkspace } from '@openmrs/esm-framework';
+import { CardHeader, EmptyState, usePatientChartStore } from '@openmrs/esm-patient-common-lib';
 import VisitSummary from './past-visits-components/visit-summary.component';
 import styles from './current-visit-summary.scss';
 
 interface CurrentVisitSummaryProps {
   patientUuid: string;
 }
 
+/**
+ * This extension is not used in the refapp.
+ * This extension uses the patient chart store and SHOULD only be mounted within the patient chart
+ */
 const CurrentVisitSummary: React.FC<CurrentVisitSummaryProps> = ({ patientUuid }) => {
   const { t } = useTranslation();
-  const { isLoading, currentVisit, error, isValidating } = useVisit(patientUuid);
+  const { patientUuid: storePatientUuid, visitContext } = usePatientChartStore(patientUuid);
 
-  if (isLoading) {
-    return (
-      <InlineLoading
-        status="active"
-        iconDescription={t('loading', 'Loading')}
-        description={t('loadingVisit', 'Loading current visit...')}
-      />
-    );
+  if (patientUuid !== storePatientUuid) {
+    return null;
   }
-
-  if (!!error) {
-    return <ErrorState headerTitle={t('failedToLoadCurrentVisit', 'Failed loading current visit')} error={error} />;
-  }
-
-  if (!currentVisit) {
+  if (!visitContext) {
     return (
       <EmptyState
         headerTitle={t('currentVisit', 'Current visit')}
```

Comment:
Unrelated to the diff but the `displayText` here should probably be: 

```
displayText={t('activeVisits__lower', 'active visits')}
```

so the full empty state text reads as "There are no active visits to display for this patient"